xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2821,6 +2822,22 @@ void CVideoPlayer::HandleMessages()
         std::shared_ptr<CInputStreamPVRBase> pvrinputstream = std::static_pointer_cast<CInputStreamPVRBase>(m_pInputStream);
         pvrinputstream->Pause(speed == 0);
       }
+      
+      // Inform an inputstream addon with a demuxer that an (un)pause has occurred
+      if (m_pInputStream->IsStreamType(DVDSTREAM_TYPE_ADDON) && speed != m_playSpeed &&
```

Comment:
this block is nonsense, same applies to the pvr related code above. videoplayer controls reading from demuxer with SetSpeed. if user pauses player and buffers are not full, player continues to read until buffers are full.

Reply:
Why is it nonsense? The api docs for pausestream simply state that a pause action has occurred. This happens for inputstreams without a demuxer but not for ones with one.

Reply:
Ok, in hindsight, I agree 👍 this time can be inferred from the demux.

Reply:
btw, wherever you see m_pInputStream->IsStreamType, you can safely assume that it is outdated crap and following this path does more harm than any good.